### ISSUE TITLE
[Question] The right way to utilize two or more proxy servers at the same time...

### ISSUE THREAD

**Issue by lss4** [shadowsocks/luci-app-shadowsocks#164]
I'm running LEDE. Apparently the LuCI part allows configuring multiple proxy profiles, but the ss-local config is strange. I can add more proxies to ss-local, but I can only define one local port.

EDIT: Apparently it turned out that there indeed were two ss-local instances running, but both are listening to the same local port, which could cause problems. For now I'll have to leave only one on until there's a solution...

EDIT 2: Edited the question once again after reading the code. From the code I believe this was done to separate local port configurations for ss_redir, ss_local and ss_tunnel, but was apparently implemented in a way that when two or more proxies are enabled, they'll all be using the same local port, which while it works, it kind of makes the traffic unpredictable (mainly in terms of contents that might be region restricted, if the proxies are of different regions). I haven't really tested such configurations, as this is not what I wanted.

I'm not certain whether this design is intended or by mistake (that the LuCI interface was made in a way to utilize only one proxy at a time).

**Comment by lss4** [comment]
Simply put... the current interface give all ss-local proxies the same local port (according to #157, the configuration file has reuse_port set to true which enabled the possibility to do so).

I believe this is done for better proxy availability when using multiple servers. However, this is not what I wanted as the two servers I'm using are for different purposes. The current interface offers no way to set local ports for each ss-local proxy at present.

**Comment by cokebar** [comment]
The purpose of this function:
1. When you are using a multi-core cpu router, eg. dual core CPU, you can open two ss-redir instance using same ss server. This will make full use of the dual core CPU, improving the performance.
2. Open several ss-redir instance using different ss server. This will provide a simple load balancing function. But surely this will cause some problem. Using haproxy or similar softeware may be a better solution.

**Comment by cokebar** [comment]
If you want to open several ss-local binding to different ports, do not use Luci-app-shadowsocks. Simply using Shell commands.

**Comment by lss4** [comment]
I'm aware of how to use shell commands but it's better off managed through UCI when using LEDE.

If you're talking about using a single remote server and utilize several instances to perform load balancing, this justifies the purpose. However, the problem is that at present, the ss-local config section _does_ allow you to utilize more than one remote servers (through the add icon), but _does not_ allow you to assign separate local ports for each of the remote servers utilized, and they'll all use the same local port once committed.

While it may possibly improve connection availability, it may result in having your IP constantly fluctuating, which can cause problems if you're logged in to certain websites that are security-conscious, as such may trip their account protection mechanisms resulting in suspicious activity warnings, or protective account supensions that require you to confirm either via email or via SMS.

**Comment by lss4** [comment]
Bump as I just tried implementing it myself. I just modified the server section to include individual local_port settings and removed the global local_port setting for the SOCKS5 section. With that I'm able to configure two separate ss-local instances, each listening to a different port and pointing to different servers.

Other parts were untouched, but consider it experimental, as I personally don't use ss-redir and cannot test whether my changes would break that. It depends on whether the local_port passed as the startup argument (outside the config.json) would override what's inside the json (if yes, then the features that I have not touched would still work as it should be).
